qdrant-landing/content/articles/new-recommendation-api.md

```diff
@@ -0,0 +1,336 @@
+the current point is better than the best match, we can stop the traversal, as this is a local minimum. We start at the biggest 
+scale, and then gradually zoom in.
+
+In this oversimplified example, we assumed that the distance between the points is the only factor that matters. In reality, we 
+might want to consider other criteria, such as the ticket price, or avoid some specific locations due to certain restrictions. 
+That means, there are various strategies for choosing the best match, which is also true in the case of vector recommendations. 
+We can use different approaches to determine the path of traversing the HNSW graph by changing how we calculate the score of a 
+candidate point during traversal. The default behaviour is based on pure distance, but Qdrant 1.6 exposes two strategies for the 
+recommendation API.
+
+### Average vector
+
+The default strategy, called `average_vector` is the previous one, based on the average of positive and negative examples. It 
+simplifies the recommendations process and converts it into a single vector search. It supports both point IDs and vectors as 
+parameters. For example, you can get recommendations based on past interactions with existing points combined with query vector 
+embedding. Internally, that mechanism is based on the averages of positive and negative examples and was calculated with the 
+following formula:
+
+```
+average_vector = avg(positive_vectors) + ( avg(positive_vectors) - avg(negative_vectors) )
+```
+
+The `average_vector` converts the problem of recommendations into a single vector search.
+
+### The new hotness - Best score
+
+The new strategy is called `best_score`. It does not rely on averages and is more flexible. It allows you to pass just negative 
+samples and uses a slightly more sophisticated algorithm under the hood.
+
+The best score is chosen at every step of HNSW graph traversal. We separately calculate the distance between a traversed point 
+and every positive and negative example. In the case of the best score strategy, **there is no single query vector anymore, but a 
+bunch of positive and negative queries**. As a result, for each sample in the query, we have a set of distances, one for each 
+sample. In the next step, we simply take the best scores for positives and negatives, creating two separate values. Best scores 
+are just the closest distances of a query to positives and negatives. The idea is: **if a point is closer to any negative than to 
+any positive example, we do not want it**. We penalize being close to the negatives, so instead of using the similarity value 
+directly, we check if it’s closer to positives or negatives. The following formula is used to calculate the score of a traversed 
+potential point:
+
+```rust
+if best_positive_score > best_negative_score {
+    score = best_positive_score
+} else {
+    score = -(best_negative_score * best_negative_score)
+}
+```
```

Comment:
Would you want to use Python instead of Rust here? Though the logic is fairly simple to read

Reply:
It's just a copy from our codebase, as far as I know. I'd keep it as it is.